src/js/router.js

```diff
@@ -1,39 +1,51 @@
+/*
+
+    The base grade for express router modules.  See the documentation for details:
+
+    https://github.com/GPII/gpii-express/blob/master/docs/router.md
+
+ */
 "use strict";
-// Base grade for express router modules...
-//
-// The express module will automatically attempt to wire in anything with this gradeName into its routing table.
-//
-// When the underlying router is created, its constructor will be passed `options.routerOptions`.  For a list of the
-// supported options, check out [the Express documentation](http://expressjs.com/api.html#router).
-//
-// This implementation is not meant to be used directly.  When wrapping most existing Express routers, you will want
-// to implement your own `router` method and (typically) point your `router` invoker at that.
-//
-// For all other use cases, you should likely start with `requestAwareRouter` and implement your own handler.
-//
 var fluid = require("infusion");
 var gpii  = fluid.registerNamespace("gpii");
 fluid.registerNamespace("gpii.express.router");
 
-// Instantiate our router object.  The root gpii.express object will wire everything together
+/**
+ *
+ * @param that {Object} - The router object itself.
+ *
+ * Instantiate our router object.  The root `gpii.express` instance will wire everything together.
+ * 
+ */
 gpii.express.router.createRouter = function (that) {
     var express = require("express");
     that.router = express.Router(that.options.routerOptions);
     that.events.routerLoaded.fire(that);
 };
 
+gpii.express.router.getRouter = function (that) {
+    return that.router;
+};
+
 fluid.defaults("gpii.express.router", {
-    gradeNames: ["fluid.component"],
-    method:     "use",
-    path:       null,
-    router:     null,
+    gradeNames:    ["gpii.express.middleware"],
     routerOptions: {},
     events: {
-        routerLoaded: null
+        routerLoaded: null,
+        onReady: {
+            events: {
+                routerLoaded: "routerLoaded"
+            },
+            args: ["{that}"]
+        }
     },
     invokers: {
-        route: {
-            funcName: "fluid.notImplemented"
+        getMiddlewareFn: {
+            funcName: "gpii.express.router.getRouter",
+            args:     ["{that}"]
+        },
+        middleware: {
+            funcName: "fluid.identity"
```

Comment:
Not a great choice since fluid.identity dispatches only one argument. Perhaps fluid.notImplemented (i.e. writing nothing) would be closer to required intent - but I'm worried since fluid.identity here suggests that there is nothing that tests this default (or tests it in a way that doesn't check that "response" is dispatched too)